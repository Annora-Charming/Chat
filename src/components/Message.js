import React from 'react';

class Message extends React.Component {
    render() {
        const { nickname, message } = this.props;
        return (
            <li>
                <b>{nickname}:</b>
                {message}
            </li>
        );
    }
}

export default Message;
