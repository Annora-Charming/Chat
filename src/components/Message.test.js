import React from 'react';
import {shallow} from 'enzyme';
import Message from './Message';
import renderer from "react-test-renderer";

test('Message shows nick and message', () => {
    const nick = 'test';
    const message = 'test';
    const component = shallow(<Message nick={nick} message={message}/>);
    expect(component.text()).toContain(nick);
    expect(component.text()).toContain(message);
});

test('Snapshot test', ()=>{
    const nick = "test";
    const message = "test";
    const component = renderer.create(<Message nick = {nick} message = {message} /> );
    expect(component.toJSON()).toMatchSnapshot();
});