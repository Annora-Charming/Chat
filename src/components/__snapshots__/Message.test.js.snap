// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test 1`] = `
<li>
  <b>
    test
    :
  </b>
  test
</li>
`;
